src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
         this.event.description = value;
     }
 
+    // This is the start time of an event instance (if recurring)
+    // See also: calendar.service.ts generateEvents
     get dtstart(): moment.Moment {
-        return this.icalTimeToLocalMoment(this.event.startDate);
+        return this.icalTimeToLocalMoment(this._dtstart);
     }
 
+    // If set on an event with RRULE, updates entire repeated event
+    // to now *start* from this new date
+    // If set on an exception, only changes that exception
     set dtstart(value: moment.Moment) {
         const allDay = this.allDay;
-        this.event.startDate = ICAL.Time.fromJSDate(value.toDate());
+        this._dtstart = this.momentToIcalTime(value, this.event.startDate ? this.event.startDate.zone : null);
+        // FIXME: Do this only if its the first one?
```

Comment:
I don't think I'm getting these two FIXMEs. How does editing a single event, down the recurrence chain, impact all the others from here?

Reply:
_dtstart (and dtstart and start) are all "start date/time of this event instance" .. event.startDate is the start of whole recurring event, if it is one.. so we shouldnt be changing that unless the object we're editing is the original start event.

I think I subsequently prevented this by disallowing "All occurences" if we were editing a non-first repeat event. (I'll add another check here just in case)

Reply:
Removed fixme and verified working correctly